cstrml/tee/src/lib.rs

```diff
@@ -370,13 +370,38 @@ impl<T: Trait> Module<T> {
 
         Used::put(total_used);
         Reserved::put(total_reserved);
+        let total_workload = total_used + total_reserved;
 
-        // 5. Call `on_report_works` handler
-        T::Works::report_works(
-            &who,
-            used + reserved,
-            total_used + total_reserved,
-        );
+        // 5. Update work report for every identity
+        let workload_map: Vec<(T::AccountId, u128)> = <TeeIdentities<T>>::iter().map(|(controller, _)| {
+            // a. calculate this controller's order file map
+            let mut order_files: Vec<(MerkleRoot, T::Hash)> = vec![];
+            if let Some(provision) = T::MarketInterface::providers(&controller) {
+                order_files = provision.file_map.values()
+                    .filter_map(|order_id| {
+                        // Get order status(should exist) and (maybe) change the status
+                        let sorder =
+                            T::MarketInterface::maybe_get_sorder(order_id).unwrap_or_default();
+                        if sorder.order_status == OrderStatus::Success {
+                            Some((sorder.file_identifier, order_id.clone()))
+                        } else {
+                            None
+                        }
+                    })
+                    .collect();
+            }
+            // b. calculate controller's own reserved and used space
+            let (reserved, used) = Self::update_and_get_workload(&controller, &order_files, rs);
```

Comment:
we do not need get this, cause this function **ONLY** be called in the new era with `update_identities`